src/main/scala/essential_scala/Ex4Traits.scala

```diff
@@ -86,24 +88,64 @@ object Ex4Traits extends App {
   // 4.6.3.2 The Forest of Trees
   // pattern matching
   sealed trait Tree {}
-  case class Node(left: Tree, right: Tree) extends Tree
-  case class Leaf(element: Int) extends Tree
-  object Math {
-    def prettyPrint(t: Tree, indent: String): Unit = t match {
-      case (t: Node) =>
-        prettyPrint(t.left, indent + " ")
-        prettyPrint(t.right, indent + " ")
-        println(indent + ".")
-      case (t: Leaf) => println(indent + t.element)
+  final case class Node(left: Tree, right: Tree) extends Tree
+  final case class Leaf(element: Int) extends Tree
+
+  object Tree {
+    def mkSpace(i: Int) = Array.fill(i) { " " }.mkString("")
```

Comment:
Why are you using braces for the second parameter of `Array.fill`? Braces are good for big enough lambdas, not for constant values.